File: adv_2022/adv_day_01/python/day_01.py

with open("input.txt") as f:
    calories = f.readlines()

biggest_calorie = 0
current_calorie = 0

for calorie in calories:
    try:
        current_calorie += int(calorie)
    except:
        if(current_calorie>biggest_calorie):
            biggest_calorie = current_calorie
        current_calorie = 0

print(biggest_calorie)
# 71023

File: adv_2022/adv_day_01/cpp/day_01.cpp
#include <iostream>
#include <fstream>

int main(){
    std::ifstream File("input.txt");
    std::string text;
    
    int biggest_calorie = 0;
    int current_calorie = 0;

    while (std::getline(File, text)) {
        try {
            current_calorie += std::stoi(text);
        }
        catch (std::exception e) {
            if (current_calorie > biggest_calorie) {
                biggest_calorie = current_calorie;
            }
            current_calorie = 0;
        }
    }
    std::cout << biggest_calorie;

    File.close();
}
// 71023